### DISCUSSION TITLE
bitbucket webhook is not working

### DISCUSSION THREAD

**Discussion by saintnoah** [argoproj/argo-cd#16142] [Q&A]
Hello,

I am trying to setup webhook for my bitbucket repo with argocd(Auto sync works perfectly fine on my argocd setup)
Followed the steps from this link: https://argo-cd.readthedocs.io/en/stable/operator-manual/webhook/

let's say argocd is deployed on http://test.foo.com

When create bitbucket webhook, url endpoint is set to: `http://test.foo.com/api/webhook`
And Test Connection return green 200

But after I pushed any commit, the sync is not been triggered on argocd.
On bitbucket webhook event, I can see response as below
![image](https://github.com/argoproj/argo-cd/assets/1174875/72f2f0b9-e752-4e69-951d-f368d1774d19)

And bitbucket webhook request header shows sending json content type with a valid body
> Headers
X-Hub-Signature: sha256=7c08ce25a328372c4ca8f31e3b260c9db90bcab14d7b4c345d3d97bfe7afc219
X-Request-Id: d8c1e2fb-c93e-454f-9e20-103df752fb62
Content-Type: application/json; charset=utf-8
X-Event-Key: repo:refs_changed

On argocd side, I tried to check logs from argocd server pod but only see below 2 lines showed up when commit was occurred

> time="2023-10-28T23:12:07Z" level=info msg="Received push event repo: ssh://git@bitbucket.bar.com:7999/san/abc-helm-charts.git, revision: argo-test, touchedHead: true"
time="2023-10-28T23:12:07Z" level=info msg="Received push event repo: https://bitbucket.bar.com/scm/san/abc-helm-charts.git, revision: argo-test, touchedHead: true"

Just wondering did I miss any configuration on bitbucket or argocd?  How can I debug further on this issue?

Thanks a lot for your help!

**Comment by ryanotella** [comment]
I don't believe this is implemented currently because it's fairly new feature of Bitbucket Cloud. It wouldn't be hard to add however.
